readthedocs/doc_builder/environments.py

```diff
@@ -362,19 +337,42 @@ def run_command_class(
                 msg += u':\n{out}'.format(out=build_cmd.output)
 
             if warn_only:
-                self._log_warning(msg)
+                log.warning(LOG_TEMPLATE.format(
+                    project=self.project.slug,
+                    version='latest',
+                    msg=msg,
+                ))
             else:
                 raise BuildEnvironmentWarning(msg)
         return build_cmd
 
 
-class LocalEnvironment(BaseEnvironment):
+class EnvironmentRecordCommandMixin(object):
+
+    # record, warn_only
+    def pre_run_command(self, cls, cmd, warn_only, kwargs):
+        kwargs.update({
+            # TODO: when is this necessary?
+            'build_env': self,
+        })
+        self.record = kwargs.pop('record', True)
+        self.record_as_success = warn_only
```

Comment:
`record_as_success` and `warn_only` are not the same.

(take a look at my previous comment where the logic to calculate these is defined)